### PR TITLE
Implement border-image-outset.

### LINKED ISSUE #16638: Support border-image-outset for gradients
~~Radial gradients are only implemented for backgrounds but [not for borders](https://github.com/pyfisch/servo/blob/49f01055b64b732fb71bdbd20306b5b9afe0cccb/components/layout/display_list_builder.rs#L1315).~~

Implement `border-image-outset` property for gradient borders.

follow up of #16385

### PR DESCRIPTION
**Pull request by pyfisch**
The border is moved correctly but linear gradients
and some images will not be scaled correctly. These
depend on border-image-slice and better webrender
support.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix (partially) #16638 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16796)

### CODE REVIEW

**Comment by pyfisch** [comment]
@glennw have you reviewed this? Does it need changes or can it be merged after a rebase?

If you don't have time to review please assign it to someone else.

Thanks for your work and have a nice weekend.

File: components/layout/display_list_builder.rs [hunk 1]
```diff
@@ -2781,6 +2793,13 @@ fn position_to_offset(position: LengthOrPercentage, Au(total_length): Au) -> f32
     }
 }
 
+fn outset_to_length(outset: LengthOrNumber, border_width: Au) -> Au {
```
Comment (emilio) [inline components/layout/display_list_builder.rs]: Let's move this to `model.rs` and call this `specified_outset`, for symmetry with other properties like border-radius.

Let's also make the doc comment of that reference https://drafts.csswg.org/css-backgrounds-3/#border-image-outset.

File: components/layout/display_list_builder.rs [hunk 2]
```diff
@@ -1180,7 +1182,8 @@ impl FragmentDisplayListBuilding for Fragment {
                                 specified(vertical, bounds.size.height)),
             EndingShape::Ellipse(LengthOrPercentageOrKeyword::Keyword(word))
                 => convert_ellipse_size_keyword(word, &bounds.size, &center),
-        };
+        } + Size2D::new((outset.left + outset.right).scale_by(0.5),
```
Comment (emilio) [inline components/layout/display_list_builder.rs]: Hmm.. Why does this modify the radius of a radial gradient? Can you quote a spec, or comment on this? I see why it's useful when it's a keyword, or a percentage, but why modifying it when there's an absolute radius?

**Review by emilio — COMMENTED** [review]
This looks fine, but needs some tests. Tests for this shouldn't be so problematic as tests for gradients, so please do add them.

I'm concerned about the gradient radius bit, could you elaborate on that?

**Comment by pyfisch** [comment]
Closing this because I am not sure anymore that this is good idea without `border-image-slice` support at the same time and the implementation of radius is problematic.

**Pull request closed without merging**